src/tests/multihost/basic/conftest.py

```diff
@@ -173,6 +175,117 @@ def disable_kcm():
+        with open('/tmp/sssd.conf', "w") as sssconf:
+            sssdconfig.write(sssconf)
+    session_multihost.master[0].transport.put_file('/tmp/sssd.conf',
+                                                   '/etc/sssd/sssd.conf')
+    session_multihost.master[0].service_sssd('restart')
+
+
+@pytest.fixture
+def enable_sudo_service(session_multihost):
+    """ Enable sudo services in sssd.conf """
+    session_multihost.master[0].transport.get_file('/etc/sssd/sssd.conf',
+                                                   '/tmp/sssd.conf')
+    sssdconfig = ConfigParser.SafeConfigParser()
+    sssdconfig.read('/tmp/sssd.conf')
+    sssdconfig.remove_option('sssd', 'services')
+    sssdconfig.set('sssd', 'services', 'nss, pam, ifp, sudo')
+    with open('/tmp/sssd.conf', "w") as sssconf:
+        sssdconfig.write(sssconf)
+    session_multihost.master[0].transport.put_file('/tmp/sssd.conf',
+                                                   '/etc/sssd/sssd.conf')
+    session_multihost.master[0].service_sssd('restart')
+
+
+@pytest.fixture
+def create_sudorule(session_multihost, create_casesensitive_posix_user):
+    """ Create posix user and groups """
+    # pylint: disable=unused-argument
+    _pytest_fixture = [create_casesensitive_posix_user]
+    ldap_uri = 'ldap://%s' % (session_multihost.master[0].sys_hostname)
+    ds_rootdn = 'cn=Directory Manager'
+    ds_rootpw = 'Secret123'
+    ldap_inst = LdapOperations(ldap_uri, ds_rootdn, ds_rootpw)
+    ldap_inst.org_unit('sudoers', 'dc=example,dc=test')
+    capsrule_dn1 = 'cn=lessrule,ou=sudoers,dc=example,dc=test'
+    capsrule_dn2 = 'cn=morerule,ou=sudoers,dc=example,dc=test'
+    sudo_attr1 = {
+        'objectClass': [b'top', b'sudoRole'],
+        'cn': 'lessrule'.encode('utf-8'),
+        'sudoHost': 'ALL'.encode('utf-8'),
+        'sudoCommand': '/usr/bin/less'.encode('utf-8'),
+        'sudoUser': 'capsuser-1'.encode('utf-8'),
+        'sudoOption': '!requiretty'.encode('utf-8')}
+    sudo_attr2 = {
+        'objectClass': [b'top', b'sudoRole'],
+        'cn': 'morerule'.encode('utf-8'),
+        'sudoHost': 'ALL'.encode('utf-8'),
+        'sudoCommand': '/usr/bin/more'.encode('utf-8'),
+        'sudoUser': 'CAPSUSER-1'.encode('utf-8'),
+        'sudoOption': '!requiretty'.encode('utf-8')}
+    no_passwd = [(ldap.MOD_ADD, 'sudoOption', '!authenticate'.encode('utf-8'))]
+    try:
+        (_, _) = ldap_inst.add_entry(sudo_attr1, capsrule_dn1)
+    except LdapException:
+        pytest.fail("Failed to add %s rule in ldap" % (capsrule_dn1))
+    try:
+        (_, _) = ldap_inst.add_entry(sudo_attr2, capsrule_dn2)
+    except LdapException:
+        pytest.fail("Failed to add %s rule in ldap" % (capsrule_dn2))
+    else:
+        (_, _) = ldap_inst.modify_ldap(capsrule_dn1, no_passwd)
+        (_, _) = ldap_inst.modify_ldap(capsrule_dn2, no_passwd)
+
+
+@pytest.fixture
+def enable_sss_sudo_nsswitch(session_multihost, tmpdir, request):
```

Comment:
Should we change the nsswitch ourselves? Shouldn't this be done with authselect/authconfig?

Reply:
authselect doesn't do it by default, Not sure if that's bug, currently authselect adds sss to passwd,group,netgroup,automount, services. 

Also "sudoers: files " entry doesn't exist in default /etc/nsswith.conf